packages/starlight/__tests__/basics/pagination-with-base.test.ts

```diff
@@ -0,0 +1,76 @@
+import { afterAll, beforeAll, describe, expect, test, vi } from 'vitest';
+import type {
+	getPrevNextLinks as getPrevNextLinksType,
+	getSidebar as getSidebarType,
+} from '../../utils/navigation';
+
+vi.mock('astro:content', async () =>
+	(await import('../test-utils')).mockedAstroContent({
+		docs: [
+			['index.mdx', { title: 'Home Page' }],
+			['environmental-impact.md', { title: 'Eco-friendly docs' }],
+			['guides/authoring-content.md', { title: 'Authoring Markdown' }],
+			['guides/components.mdx', { title: 'Components' }],
+			['reference/frontmatter.md', { title: 'Frontmatter Reference' }],
+		],
+	})
+);
+
+describe('without base', async () => {
+	let getPrevNextLinks: typeof getPrevNextLinksType;
+	let getSidebar: typeof getSidebarType;
+	let sidebar: ReturnType<typeof getSidebar>;
+
+	beforeAll(async () => {
+		({ getPrevNextLinks, getSidebar } = await import('../../utils/navigation'));
+		sidebar = getSidebar('/reference/frontmatter/', undefined);
+	});
+
+	test('pagination links are formatted correctly with no frontmatter', () => {
+		const links = getPrevNextLinks(sidebar, true, {});
+		expect(links.prev?.href).toMatchInlineSnapshot(`"/guides/components/"`);
+		expect(links.next?.href).toMatchInlineSnapshot(`undefined`);
+	});
+
+	test('pagination links are formatted correctly with custom links in frontmatter', () => {
```

Comment:
Tiny thought: should the test name not mention "formatted correctly" but more something like "used as authored" to emphasize we're not formatting the link in this case?

Reply:
Good point — updated and also switched away from snapshots to hardcoded values as I don’t think these should really change.